cellprofiler/modules/classify.py

```diff
@@ -0,0 +1,952 @@
+
+            self.path.join_parts(dir_choice, custom_path)
+
+        self.mode = cellprofiler.settings.Choice("Classify or train?", [MODE_CLASSIFY, MODE_TRAIN])
+
+        self.advanced_or_automatic = cellprofiler.settings.Choice("Configuration mode", [AA_AUTOMATIC, AA_ADVANCED],
+                                                                  doc="Do you want to automatically choose the training parameters or use the defaults?")
+
+        self.radius = cellprofiler.settings.Integer("Radius", DEFAULT_RADIUS, 1)
+
+        self.n_features = cellprofiler.settings.Integer("Number of features", DEFAULT_N_FEATURES, 1,
+                                                        doc="The classifier runs a feature reduction set. This creates <i>Eigentextures</i> which are representative texture patches found throughout the image. The module scores each patch around a pixel according to how much it has each of these textures and those scores are fed into the final classifier. Raise the number of features if some of the textures or edges of your classes are misclassified. Lower the number of features to improve processing time or to reduce overfitting if you have a smaller amount of ground truth.")
+
+        self.n_estimators = cellprofiler.settings.Integer("Number of estimators", DEFAULT_N_ESTIMATORS, 1,
+                                                          doc="The classifier uses a voting scheme where it trains this many estimators. It purposefully does a bad job training and makes up for this deficit by having many poor classification judges. This protects against overfitting by not relying on having a single classifier that is very good at classifying the ground truth, but mistakenly uses irrelevant information to do so. Raise the number of estimators if the classifier is making obvious mistakes with unwarranted certainty. Lower the number of estimators to improve processing speed.")
+
+        self.min_samples_per_leaf = cellprofiler.settings.Integer("Minimum samples per leaf", DEFAULT_MIN_SAMPLES_PER_LEAF, 1,
+                                                                  doc="This setting determines the minimum number of ground truth pixels that the classifier will use to split a decision tree. There must be at least this number of example pixels in each branch for the classifier to have confidence that the split is real and not just an artifact of an irrelevant measurement.\n Lower this setting if the classifier does a good job on most of the pixels but does not draw sharp distinctions between one class and another at the border between the classes (e.g. at the edges of cells). Raise this setting if the classifier misclassifies pixels that are clearly not the right class - this is overtraining.")
+
+        self.path = cellprofiler.settings.DirectoryPath("Classifier folder")
+
+        self.filename = cellprofiler.settings.FilenameText("Classifier file", "Classifier.cpclassifier", get_directory_fn=get_directory_fn,
```

Comment:
For get_directory_fn, a lambda function is possible, for set_directory_fn, it's a stretch and only possible because of the order of outputs from self.path.get_parts_from_path.

    set_directory_fn=lambda path: self.path.join_parts(*self.path.get_parts_from_path(path))